homeassistant/components/tado/water_heater.py

```diff
@@ -270,10 +301,13 @@ def _control_heater(self, hvac_mode=None, target_temp=None):
             self._tado.set_zone_off(self.zone_id, CONST_OVERLAY_MANUAL, TYPE_HOT_WATER)
             return
 
+        if duration:
+            overlay_mode = CONST_OVERLAY_TIMER
         # Fallback to Smart Schedule at next Schedule switch if we have fallback enabled
-        overlay_mode = (
-            CONST_OVERLAY_TADO_MODE if self._tado.fallback else CONST_OVERLAY_MANUAL
-        )
+        elif self._tado.fallback:
+            overlay_mode = CONST_OVERLAY_TADO_MODE
+        else:
+            overlay_mode = CONST_OVERLAY_MANUAL
```

Comment:
Using an else is rarely needed. Consider rewriting it without the else.

e.g.,
```python
overlay_mode = CONST_OVERLAY_MANUAL
if duration:
    overlay_mode = CONST_OVERLAY_TIMER
elif self._tado.fallback:
    overlay_mode = CONST_OVERLAY_TADO_MODE
```

Reply:
Done